src/unit_tests/os_crypto/aes/test_aes_op.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stddef.h>
+#include <stdarg.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include "../headers/shared.h"
+#include "../../os_crypto/aes/aes_op.h"
+
+// Tests
+
+void test_aes_string
+(void **state)
+{
+    const char *key = "test_key";
+    const char *string = "test string";
+    const int buffersize = 1024;
+    char buffer1[buffersize];
+    char buffer2[buffersize];
+
+    memset(buffer1, 0, sizeof(buffer1));
+    memset(buffer2, 0, sizeof(buffer2));
+
+    OS_AES_Str(string, buffer1, key, strlen(string), OS_ENCRYPT);
+    OS_AES_Str(buffer1, buffer2, key, strlen(buffer1), OS_DECRYPT);
+
+    // Remove junk from decryption
+    char *last = strchr(buffer2, 0x005);
```

Comment:
Instead of modifying the output string, you could test that the output string begins with the same content as the input string.
Maybe something like this: `assert_int_equal(strncmp(buffer2, string, strlen(string)), 0);`

Reply:
done